echo-telemetry/src/main/java/com/netflix/spinnaker/echo/telemetry/TelemetryEventListener.java

```diff
@@ -217,4 +216,61 @@ public String toString() {
       return new String(getBytes(), StandardCharsets.UTF_8);
     }
   }
+
+  // Arbitrary container for TelemetryEventListener structured data, so there aren't huge
+  // fully-qualified names for the equivalent classes in the kork.proto package all over the code.
+  public static class TEL {
+
+    @Data
+    public static class Content {
+      Execution execution = new Execution();
+    }
+
+    @Data
+    public static class Execution {
+      private String id = "";
+      private String type = "UNKNOWN";
+      private String status = "UNKNOWN";
+      private Trigger trigger = new Trigger();
+      private List<Stage> stages = new ArrayList<>();
+
+      public String getType() {
+        return type.toUpperCase();
+      }
+
+      public String getStatus() {
+        return status.toUpperCase();
+      }
+    }
+
+    @Data
+    public static class Trigger {
+      private String type = "UNKNOWN";
+
+      public String getType() {
+        return type.toUpperCase();
+      }
+    }
+
+    @Data
+    public static class Stage {
+      private String status = "UNKNOWN";
+      private String type = "UNKNOWN";
+      private String syntheticStageOwner;
+      private Context context = new Context();
+
+      public boolean isSyntheticStage() {
+        return syntheticStageOwner != null && !"".equalsIgnoreCase(syntheticStageOwner);
```

Comment:
Really minor, but `equalsIgnoreCase` is unecessary here as we're only ever comparing to the empty string.

Reply:
Dropped in favor of StringUtils.